### ISSUE TITLE
Bug: incompatible type `ParserOptions['projectFolderIgnoreList']`

### ISSUE THREAD

**Issue by nix6839** [typescript-eslint/typescript-eslint#8853]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### Repro Code

```TypeScript
None
```

### Expected Result

> **typescript-estree:** - removes the ability to supply a `RegExp` to `projectFolderIgnoreList`, and changes the meaning of the string value from a regex to a glob.
>
> -- https://github.com/typescript-eslint/typescript-eslint/blob/main/CHANGELOG.md#breaking-changes-2

`projectFolderIgnoreList` no longer supports `RegExp`, but it seems that the type has not been removed from the `@typescript-eslint/types` package:
https://github.com/typescript-eslint/typescript-eslint/blob/a7bdd1c1eb3a403c919e516d58b1d667ea23219a/packages/types/src/parser-options.ts#L73

### Actual Result

Remove `RegExp` from `@typescript-eslint/types.ParserOptions['projectFolderIgnoreList']`.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types`             | `7.5.0` |

**Comment by Josh-Cena** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/2418 changed `typescript-estree` but not the `types` package. We should update it.
